src/main.rs

```diff
@@ -33,7 +33,9 @@ async fn main() -> std::io::Result<()> {
 
     let opts: Opts = Opts::parse();
 
-    let image_dir = opts.image_dir.clone();
+    let image_dir = opts.image_dir.clone().join(&format!("{}", Utc::today()));
+    fs::create_dir_all(&image_dir)?;
```

Comment:
This will create the directory when the application starts, which means all pictures will be inside the dated directory of when the program is run rather than when a photo is taken. I think you want to move the join and directory creation to inside the photo capturing loop below so each photo will end up in the right directory without needing to restart the application at midnight.